### PR TITLE
plugin-app-back-end: Phase 1 deprecation

### PR DESCRIPTION
**Pull request by itsalam**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by itsalam** [comment]
Seems to be breaking since I removed the deprecated backend-commons package, should this be amended?

File: .changeset/gentle-vans-learn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-app-backend': patch
+---
+
+Deprecated useRouter and alpha exports and removed deprecated dependency for @backstage/backend-commons
```
Comment (Rugvip) [inline .changeset/gentle-vans-learn.md:5]:
```suggestion
Deprecated `createRouter`, alpha exports, and removed deprecated dependency on `@backstage/backend-common`
```

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! Just a couple of bits to iron out

**Comment by itsalam** [comment]
@Rugvip Can the refernces to this plugin be removed from the [backend-legacy package]?(https://github.com/backstage/backstage/blob/master/packages/backend-legacy/src/plugins/app.ts)

File: plugins/app-backend/src/index.ts [hunk 2]
```diff
@@ -19,5 +19,9 @@
  *
  * @packageDocumentation
  */
+export { appPlugin as default } from './service/appPlugin';
 
+/**
+ * @deprecated Please migrate to the new backend system as this will be removed in the future.
```
[the remark this answers is not in the document]
Comment (itsalam) [reply plugins/app-backend/src/index.ts:25]: Got it, they're already included, but just wanted to cover all cases: https://github.com/backstage/backstage/blob/9306aecd7c324e16ea3ec6515139ade62c4ff7ca/plugins/app-backend/src/service/router.ts#L107-L110

File: plugins/app-backend/package.json [hunk 3]
```diff
@@ -57,7 +57,7 @@
     "test": "backstage-cli package test"
   },
   "dependencies": {
-    "@backstage/backend-common": "^0.25.0",
+    "@backstage/backend-defaults": "workspace:^",
```
Comment (Rugvip) [inline plugins/app-backend/package.json:60]: This isn't a dependency we can have. But I think it's fine to move back, I can't see any usage?

**Review by Rugvip — COMMENTED** [review]
Yep, thank you! Just a minor note on the new dependency

File: plugins/app-backend/package.json (hunk 3 above)
Comment (itsalam) [reply plugins/app-backend/package.json:60]: You're right, I think I got it mixed up while comparing some of the other plugins,

**Comment by itsalam** [comment]
Hi sorry, let me just sign this and we can get to approving!

**Review by Rugvip — COMMENTED** [review]
Thank you! Looks good, just needs a rebase. We ended up making the change to `/alpha` across all packages so that bit is no longer needed, internal refactor is though :+1:

**Comment by itsalam** [comment]
Should be good to go! I just left the /alpha alone from what you've mentioned.

**Pull request closed without merging**